pulser-simulation/pulser_simulation/qutip_result.py

```diff
@@ -126,33 +126,64 @@ def _weights(self) -> np.ndarray:
         # Takes care of numerical artefacts in case sum(weights) != 1
         return cast(np.ndarray, weights / sum(weights))
 
+    def _get_relevant_subsystem_indices(
```

Comment:
I'm not sure any of this refactoring is needed at all, since it turns out the density matrices supported `eliminate_states()` from the start.